src/tests/ftest/deployment/agent_failure.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+from datetime import datetime
+import os
+import threading
+
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand
+from general_utils import report_errors, get_host_data
+from telemetry_utils import TelemetryUtils
+from command_utils_base import CommandFailure
+
+
+class AgentFailure(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Verify agent failure is properly handled.
+
+    :avocado: recursive
+    """
+    def get_journalctl(self, hosts, since, until, journalctl_type):
+        """Run the journalctl on the hosts.
+
+        Args:
+            hosts (list): List of hosts to run journalctl.
+            since (str): Start time to search the log.
+            until (str): End time to search the log.
+            journalctl_type (str): String to search in the log. -t param for journalctl.
+
+        Returns:
+            list: a list of dictionaries containing the following key/value pairs:
+                "hosts": NodeSet containing the hosts with this data
+                "data":  data requested for the group of hosts
+
+        """
+        command = ("sudo /usr/bin/journalctl --system -t {} --since=\"{}\" "
+                   "--until=\"{}\"".format(journalctl_type, since, until))
+        err = "Error gathering system log events"
+        results = get_host_data(
+            hosts=hosts, command=command, text="journalctl", error=err)
+
+        return results
+
+    def run_custom_ior_cmd(self, results, job_num, file_name):
```

Comment:
(pylint-arguments-differ) Number of parameters was 6 in 'IorTestBase.run_custom_ior_cmd' and is now 4 in overridden 'AgentFailure.run_custom_ior_cmd' method